wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditManager.java

```diff
@@ -0,0 +1,174 @@
+     *         if generating report is being run by other user
+     */
+    public Path generateAuditReport() throws ServerException, ConflictException {
+        if (!inProgress.compareAndSet(false, true)) {
+            throw new ConflictException("This command is being run by other user");
+        }
+
+        Path auditReport = null;
+        try {
+            String dateTime = new SimpleDateFormat("dd-MM-yyyy_hh:mm:ss").format(new Date());
+            auditReport = Paths.get(createTempDir().getAbsolutePath(), "/report_" + dateTime + ".txt");
+            Files.createFile(auditReport);
+
+            CodenvyLicense license = null;
+            try {
+                license = licenseManager.load();
+            } catch (LicenseException ignored) {
+                //Continue printing report without license info
+            }
+            reportPrinter.printHeader(auditReport, userDao.getTotalCount(), license);
+            printAllUsers(auditReport);
+        } catch (Exception exception) {
+            deleteReport(auditReport);
+            LOG.error(exception.getMessage(), exception);
+            throw new ServerException(exception.getMessage(), exception);
+        } finally {
+            inProgress.set(false);
+        }
+
+        return auditReport;
+    }
+
+    private void printAllUsers(Path auditReport) throws ServerException {
+        Page<UserImpl> currentPage = userManager.getAll(30, 0);
+        do {
+            //Print users with their workspaces from current page
+            for (UserImpl user : currentPage.getItems()) {
+                List<WorkspaceImpl> workspaces;
+                try {
+                    workspaces = workspaceManager.getWorkspaces(user.getId());
+                } catch (ServerException exception) {
+                    reportPrinter.printError("Failed to receive list of related workspaces for user " + user.getId(), auditReport);
+                    continue;
+                }
+                Map<String, List<AbstractPermissions>> wsPermissions = new HashMap<>();
+                for (WorkspaceImpl workspace : workspaces) {
+                    try {
+                        wsPermissions.put(workspace.getId(), permissionsManager.getByInstance("workspace", workspace.getId()));
+                    } catch (NotFoundException | ConflictException e) {
+                        wsPermissions.put(workspace.getId(), emptyList());
+                    }
+                }
+                reportPrinter.printUserInfoWithHisWorkspacesInfo(auditReport, user, workspaces, wsPermissions);
+            }
+            //Initialize next page if exist, otherwise stop printing report
+            final Optional<PageRef> nextPageRefOpt = currentPage.getNextPageRef();
+            if (nextPageRefOpt.isPresent()) {
+                final PageRef nextPageRef = nextPageRefOpt.get();
+                final long itemsBefore = nextPageRef.getItemsBefore();
+                if (itemsBefore < Integer.MAX_VALUE) {
```

Comment:
I think you should move this logic to another place like user manager or mb page